pkg/controller/container-runtime-config/helpers.go

```diff
@@ -244,22 +241,176 @@ func updateCRIOConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfiguratio
 	return newData.Bytes(), nil
 }
 
-func updateRegistriesConfig(data []byte, internalInsecure, internalBlocked []string) ([]byte, error) {
-	tomlConf := new(tomlConfigRegistries)
-	if _, err := toml.Decode(string(data), tomlConf); err != nil {
+// prefixMatchesRegistry returns true if a prefix value (as in sysregistriesv2.Registry.Prefix / sysregistriesv2.Endpoint.Location)
+// matches a host[:port] value in reg.
+func prefixMatchesRegistry(prefix, reg string) bool {
+	// NOTE: "prefix" is the sysregistries.Registry name of the object, and it can be a host[:port] or a namespace within;
+	// i.e. we want to return true if "reg" is a prefix of "prefix"!
+	if reg == prefix {
+		return true
+	}
+	if len(prefix) > len(reg) {
```

Comment:
sorry but could you explain what this comparision is doing? ive read it a few times and dont fully get it?

Reply:
`prefix` is a scope for the configuration, which can be a host[:port] or host[:post]/namespace[/namespace…], or a host[:port]/namespaces…/image (or, ultimately, include a tag and/or digest, but we don’t generate such configs in this controller).

`reg` is a host[:port] value, and the purpose of this function is to match all scopes (i.e. `prefix` values) that use the host[:port] value of `reg`. 
e.g.

`reg` | `prefix` | result
--- | ------ | ------
`example.com` | `quay.io` | false
`quay.io` | `quay.io` | true
`quay.io:443` | `quay.io` | false
`quay.io` | `quay.io/namespace` | true
`quay.io` | `quay.io/rhel8/rhel` | true

Yes, the naming is confusing, as the comment says, I’d appreciate any suggestion.

Reply:
if we check the reg==prefix and return true, is there anyway to just fold that into the strings.HasPrefix logic? Because HasPrefix returns true if reg==prefix as well

https://play.golang.org/p/ZJFrQhNErbt

Just wondering if we need that '/' check? And trying to wrap my brain around this PR ;)

Reply:
actually disregard as im seeing the ':' is a possibility. thanks for the chart.

Reply:
@mtrmac ok im a little confused about line 4 of your chart bc that seems like it would yield false? in your example reg (quay.io) will return False for HasPrefix quay.io/namespace that and since it's an && it will return false there...
https://play.golang.org/p/zZWdgopC4WI

Reply:
wait i see it. HasPrefix(prefix, reg) not (reg, prefix)... Going to try to think of a new name for prefix here bc i think it's confsing me...

Reply:
if prefix = `quay.io` and reg = `quay.io/somethingelse` this will return false but that's a textbook prefix. 

we are really checking if reg is in (as a prefix lol) prefix so maybe rename prefix to sysregistriesRegistryName or sysregistriesName and eliminate the prefix wording altogether??

Reply:
> actually disregard as im seeing the ':' is a possibility. thanks for the chart.

Not just that, it’s also making sure that `quay.io.example.com` and `quay.ioprefix` don’t match `quay.io`.

Reply:
> we are really checking if reg is in (as a prefix lol) prefix so maybe rename prefix to `sysregistriesRegistryName` or `sysregistriesName` (taken from leading comments before func) or `configScope` and eliminate the prefix wording altogether??

Thanks, `scope` does express this much better. (I’ll also add unit tests to record some of the aspects of the behavior.)